src/Compilers/Core/Portable/DiagnosticAnalyzer/CompilationWithAnalyzersOptions.cs

```diff
@@ -48,6 +49,14 @@ public sealed class CompilationWithAnalyzersOptions
         /// </summary>
         public bool ReportSuppressedDiagnostics => _reportSuppressedDiagnostics;
 
+        /// <summary>
+        /// Callback to allow individual analyzers to have their own <see cref="AnalyzerOptions"/> distinct
+        /// from the shared instance provided in <see cref="Options"/>.  If <see langword="null"/> then <see cref="Options"/>
+        /// will be used for all analyzers.
+        /// </summary>
+        public Func<DiagnosticAnalyzer, AnalyzerOptions>? AnalyzerSpecificOptionsFactory
```

Comment:
note: this prop doesn't need to be public (though that matches the reest here).  The API review covers if we should expose this (since it was a constructor param) or if we should keep it internally since we hvve no actual public consumers.